tests/iscsi/iscsi_client.pm

```diff
@@ -16,50 +16,145 @@ use strict;
+            after_reboot  => {start_on_boot => 'alt-a'}
+        );
+    }
+    # go to initiator name field
+    send_key "alt-i";
+    type_string_slow_extended($test_data->{initiator_conf}->{name} . ':' . $test_data->{initiator_conf}->{id});
     assert_screen 'iscsi-initiator-service';
-    send_key "alt-v";             # go to discovered targets tab
+}
+
+sub initiator_discovered_targets_tab {
+    # go to discovered targets tab
+    send_key "alt-v";
     assert_screen 'iscsi-discovered-targets', 120;
-    send_key "alt-d";             # press discovery button
+    # press discovery button
+    send_key "alt-d";
     assert_screen 'iscsi-discovery';
-    send_key "alt-i";             # go to IP address field
-    wait_still_screen(2, 10);
-    type_string "10.0.2.1";
+    # go to IP address field
+    send_key "alt-i";
+    my $target_ip_only = (split('/', $test_data->{target_conf}->{ip}))[0];
+    type_string_slow_extended $target_ip_only;
     assert_screen 'iscsi-initiator-discovered-IP-adress';
-    send_key "alt-n";                                     # next
-    assert_and_click 'iscsi-initiator-connect-button';    # press connect button
+    # next and press connect button
+    send_key "alt-n";
+    assert_and_click 'iscsi-initiator-connect-button';
     assert_screen 'iscsi-initiator-connect-manual';
-    send_key "alt-n";                                     # go to connected targets tab
+    send_key 'alt-o';
+    assert_screen 'iscsi-initiator-discovery-enable-login-auth';
+    send_key 'alt-u';
+    type_string_slow_extended $test_data->{initiator_conf}->{user};
+    assert_screen 'iscsi-initiator-discovery-auth-initiators-username';
+    send_key 'alt-p';
+    my $init_pass = reverse $test_data->{common}->{password};
+    wait_screen_change { type_string_slow_extended $init_pass; };
+    send_key 'alt-r';
+    type_string_slow_extended $test_data->{target_conf}->{user};
+    assert_screen 'iscsi-initiator-discovery-auth-targets-username';
+    send_key 'alt-a';
+    wait_screen_change { type_string_slow_extended $test_data->{common}->{password}; };
+    send_key 'alt-n';
+}
+
+sub initiator_connected_targets_tab {
+    # go to discovered targets tab
+    send_key "alt-d";
     assert_screen 'iscsi-initiator-discovered-targets';
-    send_key "alt-n";                                     # next
+    # go to connected targets tab
+    send_key "alt-n";
     assert_screen 'iscsi-initiator-connected-targets';
-    send_key "alt-o";                                     # OK
+    # press OK twice
+    send_key "alt-o";
     wait_still_screen(2, 10);
+    send_key "alt-o";
+}
+
+
+sub run {
+    prepare_xterm_and_network;
+    mutex_wait('iscsi_target_ready', undef, 'Target configuration in progress!');
+    record_info 'Target Ready!', 'iSCSI target is configured, start initiator configuration';
+    my $module_name = y2_module_consoletest::yast2_console_exec(yast2_module => 'iscsi-client');
```

Comment:
As we use X11 module here, shouldn't we use `launch_yast2_module_x11` instead?

Reply:
Xterm has been already opened because of static network configuration. I have rather kept it opened and keep on using it until the end of the whole test suite, than to close it and open it only in case of need.